rfcs/0018-glides-trainsheet-events.md

```diff
@@ -0,0 +1,302 @@
+### Trip dropped
+This removes a trip (either added or scheduled), indicating that it will not be running.
+
+Event type: `com.mbta.ctd.glides.trip-dropped.v1`
+Fields in the event:
+- `actor` (Actor): the inspector adding the trip
+- `tripKey` (Trip Key): which trip to drop. can be either an added or scheduled trip.
+- `reason` (string): a text description of why the trip is being dropped
+### Trip restored
+This restores a trip (either added or scheduled) that was previously dropped.
+
+Event type: `com.mbta.ctd.glides.trip-restored.v1`
+Fields in the event:
+- `actor` (Actor): the inspector adding the trip
+- `tripKey` (Trip Key): which trip to drop. can be either an added or scheduled trip.
+### Trip updated
+This indicates that a trip was updated in some fashion:  consist, operators, departure time. Fields which are not present are not considered to be updated.
+
+Event type: `com.mbta.ctd.glides.trip-updated.v1`
+Fields in the event:
+- `actor` (Actor): the inspector adding the trip
+- `tripKey` (Trip Key): which trip is being updated
+- `departureTime` (HH:MM:SS time relative to the service day, optional): if present, the new time that the train is expected to depart
+- `consist` (array of Optional Car, optional): the cars assigned to perform this trip. If a car is `null`, then no car is currently assigned to that position in the consist. The front car should be listed first.
+- `operators` (array of Optional Operator, optional): the list of operators assigned to perform this trip. If an operator is `null`, then no operator is assigned to the car in that position in the consist.
+
+# Reference-level explanation
+## Effects on RTR
+These inputs should be similar to what RTR is already receiving from OCS.
+- Trip added: `TSCH NEW`
+- Trip dropped: `TSCH DEL` with delete status 1
+- Trip restored: `TSCH DEL` with delete status 0
+- Trip updated: `TSCH CON` (set consist), `TSCH OFF` (change departure time)
+
+One key difference is that the trainsheets do not record the destination for trips, so added trips only include one stop. RTR should assume a default pattern for these trips.
+
+## Long-term storage / querying
+For future use, events will be archived to S3, using Kinesis Firehose. They can either be sent to LAMP's folder, or a new folder.
+
+## Examples
+### Train departing 15 minutes later than scheduled
+- Inspector Alice (badge number: 123) is working at Mattapan
+- The 1:30am trip is being held 15 minutes to wait for a final Red Line train
+```json
+{
+  "specversion": "1.0",
+  "type": "com.mbta.ctd.glides.trip-updated.v1",
+  "source": "glides.mbta.com",
+  "id": "19fdb184-7dd6-4664-8472-04bd6177ec44",
+  "time": "2023-01-23T01:25:00-05:00",
+  "data": {
+    "actor": {
+      "username": "ainspector",
+      "badgeNumber": "123"
+    },
+    "tripKey": {
+      "serviceDate": "2023-01-22",
+      "stop": "place-matt",
```

Comment:
So the id used here is a parent station, rather than a child platform id? That should be written in the spec. We could also consider the name `station` instead of `stop` to imply that.

What about events that happen at non-stop locations? For example, an operator who signs in at Inner Belt. Glides is considering using custom internal ids (they'd look like `yard-ibelt`) for stuff that's not in GTFS. ODS also includes nonrevenue facility ids that could be supported in these fields.

Reply:
I'll clarified that it can be either a GTFS station or stop.

Is there anything written up about having Glides trainsheets used at non-terminal stations?

Reply:
Not yet, it's a future concern that we know we'll have to figure out later but we haven't started working on it yet.

Reply:
Made it more generic as a "location" so it should be able to handle the Inner Belt case now.